### PR TITLE
260/admin application edits

### PR DESCRIPTION
**Pull request by tomlovesgithub**
## What's included?
This PR asserts a new pattern for the `application.vue`  grouping it into sections which contain `editableField.vue`s.
These changes for now are only limited to the Personal Details and Character information sections, the intention of this PR is for the code patterns and basic functionality/UI decisions to be reviewed. 

## Who should test?
✅ Product owner
✅ Developers
✅ UTG

### Feature only available to auth users
- Go to an exercise.
- View an application
- Click 'Edit Application'
- Observe changes on the page;
  - 'Change' links appear
  - Button changes to 'Done editing' 
  - 'Add' and 'Remove' buttons for list sections.

## Risk - how likely is this to impact other areas?
🔴 High risk - this includes a lot of changes to shared code

## Additional context
Include screen grabs, video demo, notes etc.
image
image
image

image

## Video

https://user-images.githubusercontent.com/44227249/144861329-881c76ba-c020-4e4d-9433-01676b9683e1.mov

---
PREVIEW:DEVELOP
_can be OFF, DEVELOP or STAGING_

### CODE REVIEW

**Review by warrensearle — DISMISSED** [review]
@tomlovesgithub this PR appears to include lots of file changes
image

did something go wrong with a merge/rebase? we need to sort it out before merging this PR

**Comment by mbrookeswebdev** [comment]
@tomlovesgithub It looks like there are some failing tests + should /QualifyingTest/Response/View.vue and QualifyingTests/QualifyingTest/View.vue be removed from your PR?

**Comment by mbrookeswebdev** [comment]
It seems there is quite a lot of unused space in that table. Wouldn't be easier to place the fields underneath, for example if editMode = true, then plug in the components from Apply to input and save data, if editMode = false, show Summary components existing in Admin or take something from Apply and adopt? Just my 2 pence... :) 

Edits3
Edits

**Review by HalcyonJAC — CHANGES_REQUESTED** [review]
I've highlighted some things - in this MR there's quite a bit of commented code...

Also I can't get this to run so can't function test it, I'm getting this, but I don't think its this MR doing it, and the error is nonsensical.

image

File: src/views/Exercise/Tasks/QualifyingTests/QualifyingTest/View.vue [hunk 1]
```diff
@@ -268,14 +268,14 @@
         Open tests
       </ActionButton>
 
-      <button
+      <!-- <button
```
[the remark this answers is not in the document]
Comment (tomlovesgithub) [reply src/views/Exercise/Tasks/QualifyingTests/QualifyingTest/View.vue:271]: It never could! 
Button was annoying me as it is non-functional and should be removed.
Don't think this change should necessarily be within this PR tho so ill shall return it to before

File: src/store/candidates.js [hunk 2]
```diff
@@ -63,6 +63,14 @@ export default {
       const ref = collection.doc(`${id}/documents/personalDetails`);
       await ref.update(data);
     },
+    // saveCharacterInfo: async (context, { data, id }) => {
```
[the remark this answers is not in the document]
Comment (tomlovesgithub) [reply src/store/candidates.js:66]: No! Good catch, this was a early mistake i picked up on just before i opened the PR, commented it out to double check it wasn't needed, removed now!

**Review by HalcyonJAC — APPROVED** [review]
Great - thanks for making those changes 👍 :)

**Comment by warrensearle** [comment]
Hi @tomlovesgithub 

Upon using the preview URL I observed the following issues:

No editor form for prefessional background:
image

Location preferences editor form works for Single Choice but not for Multi or Ranked Choices
image

Experience titles are field names rather than proper language
image

Additional selection criteria 'words' layout needs improvement (see that the content appears lower than the 'words' label and the 'change' link)
image

Qualifications & Memberships are not editable or are being displayed when they shouldn't be
image

File: src/views/Candidates/CandidatesView.vue [hunk 3]
```diff
@@ -92,14 +92,14 @@ import PersonalDetails from '@jac-uk/jac-kit/draftComponents/Candidates/Personal
...
-import CharacterDeclarationSummary from '@/views/InformationReview/CharacterDeclarationSummary';
-import CharacterInformationSummaryV1 from '@/views/Exercise/Applications/CharacterInformationSummaryV1.vue';
+import CriminalOffencesSummary from '@/views/InformationReview/CharacterReview/CriminalOffencesSummary';
+import FixedPenaltiesSummary from '@/views/InformationReview/CharacterReview/FixedPenaltiesSummary';
+import MotoringOffencesSummary from '@/views/InformationReview/CharacterReview/MotoringOffencesSummary';
+import FinancialMattersSummary from '@/views/InformationReview/CharacterReview/FinancialMattersSummary';
+import ProfessionalConductSummary from '@/views/InformationReview/CharacterReview/ProfessionalConductSummary';
+import FurtherInformationSummary from '@/views/InformationReview/CharacterReview/FurtherInformationSummary';
+import CharacterDeclarationSummary from '@/views/InformationReview/CharacterReview/CharacterDeclarationSummary';
+import CharacterInformationV1Summary from '@/views/Exercise/Applications/CharacterInformationV1Summary.vue';
```
Comment (warrensearle) [inline src/views/Candidates/CandidatesView.vue:102]: @tomlovesgithub why did you change the name of this file?
imo `CharacterInformationSummaryV1` is better than `CharacterInformationV1Summary`.
With `CharacterInformationV1Summary` we risk losing the version information `V1`

File: src/views/Exercise/Details/Overview.vue [hunk 4]
```diff
@@ -27,13 +27,13 @@
           <div class="govuk-grid-column-one-half">
             <p class="govuk-body">
               Draft
-              <span class="govuk-heading-l govuk-!-margin-top-1">{{ exercise.applications.draft || 0 }}</span>
+              <!-- <span class="govuk-heading-l govuk-!-margin-top-1">{{ exercise.applications.draft || 0 }}</span> -->
```
Comment (warrensearle) [inline src/views/Exercise/Details/Overview.vue:30]: ??

File: src/views/InformationReview/AssessmentsSummary.vue [hunk 5]
```diff
@@ -0,0 +1,344 @@
...
+        objChanged[obj.index] = obj.change;
+      } else if (obj.hasOwnProperty('index') && obj.remove) { // REMOVE
+        if (objChanged.length > 0){
+          objChanged.splice(obj.index, 1);
+        } else {
+          objChanged = [];
+        } 
+      } else {
+        objChanged = obj;
+      }
```
Comment (warrensearle) [inline src/views/InformationReview/AssessmentsSummary.vue:326]: This is repeated in a few files please refactor.

https://github.com/jac-uk/admin/pull/1459#discussion_r729747511

**Review by warrensearle — DISMISSED** [review]

**Review by warrensearle — APPROVED** [review]
Lots of changes here!
✅ Code reviewed
✅ Tested (lightly) via Preview URL

**Comment by warrensearle** [comment]
@tomlovesgithub can you fix conflicts and then merge this

**Review by joy-ade — APPROVED** [review]
Positive test by product Team

**Review by HalcyonJAC — APPROVED** [review]
The date field allows invalid dates to be entered. Mainly the year part of the input would perhaps benefit from further validation.  I would suggest not allowing a year that is before 1900, or allow the programmer to specify a min-date allowed. Note: From memory, I think added a min-date validation option to the <Field> widget  at some point.

image

**Comment by joy-ade** [comment]
@tomlovesgithub Can you make personal details section editable for the Administrators please.

**Review by warrensearle — APPROVED** [review]

**Pull request merged**